orc8r/cloud/go/services/tenants/servicers/protected/servicer.go

```diff
@@ -124,6 +124,37 @@ func (s *tenantsServicer) CreateOrUpdateControlProxy(c context.Context, request
 	return &tenant_protos.CreateOrUpdateControlProxyResponse{}, nil
 }
 
+func (s *tenantsServicer) GetControlProxyFromNetworkID(_ context.Context, request *tenant_protos.GetControlProxyFromNetworkIDRequest) (*tenant_protos.GetControlProxyResponse, error) {
+	tenantList, err := s.store.GetAllTenants()
+	if err != nil {
+		return nil, err
+	}
+
+	var tenantID int64
+	isTenantFound := false
+	for _, t := range tenantList.GetTenants() {
+		for _, n := range t.Tenant.Networks {
+			if n == request.GetNetworkID() {
+				tenantID = t.Id
+				isTenantFound = true
+				break
```

Comment:
I am never sure how a `break` that is in two nested loops behaves. Can you make that clearer here?

Reply:
I moved the loops into a separate function to resolve this issue.